packages/typescript-estree/src/convert.ts

```diff
@@ -2981,13 +2985,17 @@ export class Converter {
     }
   }
 
-  #throwErrorIfDeprecatedPropertyExists<Node extends ts.Node>(
-    node: Node,
-    property: unknown,
-    message: string,
-  ): void {
-    if (property) {
-      throw createError(this.ast, node.pos, message);
+  #throwError(node: ts.Node | number, message: string): void {
```

Comment:
```suggestion
  #throwError(
    node: ts.Node | number,
    message: string,
  ): asserts node is never {
```

with that you cna remove all type casts that you added eg,

```diff
- property: this.convertJSXIdentifier(node.name),
+ property: this.convertJSXIdentifier(node.name as ts.Identifier),
```

Reply:
Thanks!